packages/import-bundle/src/index.js

```diff
@@ -98,17 +101,27 @@ export async function importBundle(bundle, options = {}, powers = {}) {
     // `filePrefix`, which will be used as the sourceMap for the top-level
     // module. The sourceMap name for other modules will be derived from
     // `filePrefix` and the relative import path of each module.
-    endowments.nestedEvaluate = src => c.evaluate(src);
+    endowments.nestedEvaluate = src => compartment.evaluate(src);
+    source = `(${source})\n${sourceMap || ''}`;
+  } else if (moduleFormat === 'endoScript') {
+    // The 'endoScript' format is just a script.
   } else {
     Fail`unrecognized moduleFormat '${moduleFormat}'`;
   }
 
-  c = new CompartmentToUse(endowments, {}, { transforms });
-  harden(c.globalThis);
-  const actualSource = `(${source})\n${sourceMap || ''}`;
-  const namespace = c.evaluate(actualSource)(filePrefix);
-  // namespace.default has the default export
-  return namespace;
+  compartment = new CompartmentToUse(endowments, {}, { transforms });
+  harden(compartment.globalThis);
+  const result = compartment.evaluate(source);
+  if (moduleFormat === 'endoScript') {
+    // The completion value of an 'endoScript' is the namespace.
+    // This format does curry the filePrefix.
```

Comment:
s/does/does not/ ? or did you mean the format holds the filePrefix internally somehow?